practice-areas/engineering/git.md

```diff
@@ -147,6 +147,151 @@ Here is a recommended Git configuration block that it is recommended to add to y
   ff = only
 [pull]
   ff = only
+[commit]
+  gpgsign = true
+```
+
+## Commit Signing
+
+Commit signing allows for identification and non-repudiation of individual commits to repositories in both Github and Gitlab. It improves security of our repositories by ensuring we know who made changes. The following instructions describe how to create a new GPG keypair, associate it with your Github and Gitlab accounts, and enable commit signing by default with git.
+
+### Creating a new GPG Key
+
+**Note: It is possible to use an SSH keypair to sign commits, but these instructions cover using GPG keys.**
+
+**OS X Users NOTE:** If the instructions below do not work for you, you may need to upgrade your copy of gpg. To update try: https://gpgtools.org It is recommended that you customize the installation of gpgTools to NOT include gpg Mail (you can deselect it if you customize the installation)
+
+[Github instructions](https://docs.github.com/en/authentication/managing-commit-signature-verification/generating-a-new-gpg-key)
+
+```shell
+gpg --full-generate-key
+```
+
+-   Select "`(1) RSA and RSA (default)`" as the type.
+-   Select a 4096 bit keysize "`What keysize do you want? (3072) 4096`"
+-   Set the key to not expire "`Key is valid for? (0)`"
+-   Set your full name and the email address associated with your Github and Gitlab accounts:
+
+```shell
+Real name: first-name last-name
+Email address: user@example.com
+Comment:
+You selected this USER-ID:
+    "first-name last-name <user@example.com>"
+```
+
+-   Set a complex passphrase for your private key.
+
+It should return: "`public and secret key created and signed.`" and the new keypair should be present under `.gnupg/` in your home directory.
+
+**Note:** TBD what if we use a different email address for Github vs Gitlab? Does it have to match?
+
+List your keys and make note of the value after `sec  rsa4096/`. This will be used for specifying which key you are working with. **OSX User Note:** You can use the email address you used to create your key if your key build result did not include a value after the `sec  rsa4096/` notice.
+
+```shell
+gpg --list-secret-keys --keyid-format=long
+
+gpg: checking the trustdb
+gpg: marginals needed: 3  completes needed: 1  trust model: pgp
+gpg: depth: 0  valid:   1  signed:   0  trust: 0-, 0q, 0n, 0m, 0f, 1u
+/home/username/.gnupg/pubring.kbx
+-------------------------------
+sec   rsa4096/FF94245366AB39B2 2022-09-13 [SC]
```

Comment:
Even though it is just a fingerprint, consider randomizing the values here (and maybe blur it out in the assets above). You might want to blur our the "mcgee" part of your name, too -- this is a public page. Up to you (unless there is Guidebook guidance on this.)

Reply:
@sfmcgee did you want to consider any of @openprivacy's comments before we merge this?

Reply:
@ContentBrewmaster Yes, actually I would, thank you. I just haven't had a chance to work on those updates yet.

Reply:
@openprivacy - thanks for the suggestion, I updated the fingerprints and the images.